craft_parts/sources/git_source.py

```diff
@@ -263,12 +278,21 @@ def _clone_new(self):
                 "checkout",
                 self.source_commit,
             ]
+            logger.debug("Executing: %s", " ".join([str(i) for i in command]))
             self._run(command)
 
     def is_local(self) -> bool:
         """Verify whether the git repository is on the local filesystem."""
         return Path(self.part_src_dir, ".git").exists()
 
+    def _is_source_local(self) -> bool:
+        """Verify if source is a filepath.
+
+        :return: True if source is a valid filepath,
+        """
+        pattern = re.compile(r"^[\w\-. \/]+$")
```

Comment:
Maybe check for absence of a scheme and see if the path is a local directory?

Reply:
Sure, I added the checking for the absence of a protocol and simplified the code a bit more.

I'm on the fence about checking if the local path exists.  First, Git's error handling for non-existent paths is good.  Second, if it's a file path to a networked location, it's going to add a delay.

Reply:
Just checking the protocol should be good.